src/lib/import_export/csv_parser.cpp

```diff
@@ -13,23 +13,26 @@
 #include "import_export/csv_meta.hpp"
 #include "resolve_type.hpp"
 #include "scheduler/job_task.hpp"
+#include "storage/dictionary_compression.hpp"
 #include "storage/table.hpp"
 #include "utils/assert.hpp"
 #include "utils/load_table.hpp"
 
 namespace opossum {
 
-std::shared_ptr<Table> CsvParser::parse(const std::string& filename, const std::optional<CsvMeta>& csv_meta) {
+std::shared_ptr<Table> CsvParser::parse(const std::string& filename, const std::optional<CsvMeta>& csv_meta,
+                                        const bool auto_compress) {
   // If no meta info is given as a parameter, look for a json file
   if (csv_meta == std::nullopt) {
     _meta = process_csv_meta_file(filename + CsvMeta::META_FILE_EXTENSION);
   } else {
     _meta = csv_meta.value();
   }
-  return _parse(filename);
+
+  return _parse(filename, auto_compress);
 }
 
-std::shared_ptr<Table> CsvParser::_parse(const std::string& filename) {
+std::shared_ptr<Table> CsvParser::_parse(const std::string& filename, const bool auto_compress) {
```

Comment:
I'm just seeing this now. Having two methods `parse` and `_parse` is weird. Do we need the `_parse` one or could we put both into `parse`?

Reply:
True, this was a leftover from when I had two public `CvsParser::parse()` methods with different signatures for convenience, but that is gone anyway. Thanks for pointing that out.